doc/source/serve/tutorials/deploy-ml-frameworks.md

```diff
@@ -0,0 +1,243 @@
+```console
+$ serve run tutorial_tensorflow:app
+```
+
+Let's query it! Open up an interactive Python shell or run the following in a separate Python script:
+
+```python
+import requests
+import numpy as np
+
+resp = requests.get(
+    "http://localhost:8000/mnist", json={"array": np.random.randn(28 * 28).tolist()}
+)
+print(resp.json())
+```
+
+You should get an output like the following (the exact prediction may vary):
+
+```bash
+{
+ "prediction": [[-1.504277229309082, ..., -6.793371200561523]],
+ "file": "/tmp/mnist_model.h5"
+}
+```
+
+
+:::{tabbed} Pytorch
+
+## Pytorch Tutorial
+
+In this guide, we will load and serve a PyTorch Resnet Model.
+In particular, we show:
+
+- How to load the model from PyTorch's pre-trained modelzoo.
+- How to parse the JSON request, transform the payload and make a prediction.
+
+This tutorial will require PyTorch and Torchvision. Ray Serve is framework agnostic and works with any version of PyTorch. If you don't already have them installed, run the following:
+
+```console
+$ pip install torch torchvision
+```
+
+Open a new Python file called `tutorial_pytorch.py`. First, let's import Ray Serve and some other helpers.
+
+```{literalinclude} ../../../../python/ray/serve/examples/doc/tutorial_pytorch.py
+:end-before: __doc_import_end__
+:start-after: __doc_import_begin__
+```
+
+We define a class `ImageModel` that parses the input data, transforms the images, and runs the ResNet18 model loaded from `torchvision`. It is decorated with `@serve.deployment` to make it a deployment object so it can be deployed onto Ray Serve. Note that the Serve deployment is exposed over an HTTP route, and by default the `__call__` method is invoked when a request is sent to your deployment over HTTP.
+
+> **NOTE**: When `ImageModel` is deployed and instantiated, it will load the resnet18 model from `torchvision` so that it can be ready to run inference on the model and serve requests later.
+
+```{literalinclude} ../../../../python/ray/serve/examples/doc/tutorial_pytorch.py
+:end-before: __doc_define_servable_end__
+:start-after: __doc_define_servable_begin__
+```
+
+Now that we've defined our Serve deployment, let's prepare it so that it can be deployed.
+
+```{literalinclude} ../../../../python/ray/serve/examples/doc/tutorial_pytorch.py
+:end-before: __doc_deploy_end__
+:start-after: __doc_deploy_begin__
+```
+
+> **NOTE**: `ImageModel.bind()` returns a `DeploymentNode` object (wrapping an `ImageModel` deployment object) that can then be used to connect with other `DeploymentNodes` to form a more complex [deployment graph](deployment-graph-e2e-tutorial).
+
+Finally, we can deploy our model to Ray Serve through the terminal.
+```console
+$ serve run tutorial_pytorch:app
+```
+
+Let's query it! Open up an interactive Python shell or run the following in a separate Python script:
+
+```python
+import requests
+
+ray_logo_bytes = requests.get(
+    "https://github.com/ray-project/ray/raw/"
```

Comment:
Is there any way to add a commit hash to this link, so it still works if the image is removed in the future?

Reply:
I've uploaded it to the [images](https://github.com/ray-project/images) repo!